lib/components/wallet_connect/approve_sign_transaction.dart

```diff
@@ -50,22 +53,91 @@ class ApproveSignTransaction extends StatefulWidget {
 class _ApproveSignTransactionState extends State<ApproveSignTransaction> {
   final ApplicationStore appStore = getIt<ApplicationStore>();
   final WalletConnectService wcService = getIt<WalletConnectService>();
-  final _liveApi = getIt<QubicLiveApi>();
+  final wCModalsController = WalletConnectModalsController();
+  final _globalSnackBar = getIt.get<GlobalSnackBar>();
+
   bool hasAccepted = false;
   String? toIdName;
   bool isLoading = false;
   @override
   void initState() {
     super.initState();
 
-    var item = appStore.currentQubicIDs.where((e) => e.publicId == widget.toID);
+    var item =
+        appStore.currentQubicIDs.where((e) => e.publicId == widget.data.toID);
```

Comment:
I've seen this code in multiple part in the app. It would be great to create a findById method in the appStore usable across the whole app.